src/recordFactory.js

```diff
@@ -1,10 +1,14 @@
-const updateRecord = (collectionName, record, database) => {
-  const originalRecord = database.find(collectionName, {id: record.id});
+import _ from 'lodash';
+
+const updateRecord = (record, collectionName, store) => {
+  const originalRecord = _.find(store[collectionName], {id: record.id});
   return Object.assign(originalRecord, record);
 };
 
-export const recordFactory = (record, collectionName, database) => {
-  record.save = () => updateRecord(collectionName, record, database);
+export const recordFactory = (record, collectionName, store) => {
+  record.save = function() {
```

Comment:
because save() method must be scoped to the object that it's invoked on, arrow function is always bind to original record ;)